src/dune_engine/dep.ml

```diff
@@ -98,16 +100,18 @@ module Fact = struct
     let paths t = t.files
 
     let make ~files ~dirs =
-      let parent_dirs =
+      let dirs_that_must_exist =
         let f path (_ : Digest.t) acc =
-          Path.Set.add acc (Path.parent_exn path)
+          match Path.as_in_build_dir path with
+          | None -> acc
```

Comment:
This used to be an error but now we silently ignore a build path. Should we raise a `Core_error` instead?

Reply:
Actually, hand on, maybe it wasn't an error. I guess I don't understand how it worked before...